daffodil-test/src/test/scala/org/apache/daffodil/runtime1/layers/IPv4ChecksumLayer.scala

```diff
@@ -36,38 +36,42 @@ final class IPv4ChecksumLayer()
   extends ChecksumLayer(
     "IPv4Checksum",
     "urn:org.apache.daffodil.layers.IPv4Checksum",
-    20,
   ) {
 
-  /**
-   * Result DFDL variable value getter. This getter is called to obtain the value for,
-   * and populate the DFDL variable named checkDigit, in the layer's target namespace.
-   * @return the check digit value
-   */
-  def getDFDLResultVariable_IPv4Checksum: JInt =
-    getChecksum
-
   /**
    * This layer is always exactly 20 bytes long.
    */
   private def lenInBytes = 20
   private def chksumShortIndex = 5
 
+  override protected def init(lr: LayerRuntime): Unit = {
+    setLength(lenInBytes)
+    super.init(lr)
+  }
+
+  /**
+   * Result DFDL variable value getter. This getter is called to obtain the value for,
+   * and populate the DFDL variable named checkDigit, in the layer's target namespace.
+   * @return the check digit value
+   */
+  def getLayerVariableResult_IPv4Checksum: JInt = getChecksum
+
   /**
    * Computes the checksum value.
    * When unparsing this also modifies the output bytes to have the checksum at the
    * middle location as per the IPv4 spec.
    * The LayerChecksumMixin assigns the computed checksum value to the first variable.
-   * @param layerCompileInfo structure providing access to state such as variables, when needed.
+   * @param layerRuntime structure providing access to state such as variables, when needed.
    * @param isUnparse true if the checksum is for unparsing.
    * @param byteBuffer contains the 20 bytes to be used to compute this checksum.
    * @return the computed checksum value as a 32-bit signed Int.
    */
   override def compute(
-    layerCompileInfo: LayerRuntime,
+    layerRuntime: LayerRuntime,
     isUnparse: Boolean,
     byteBuffer: ByteBuffer,
   ): Int = {
+    init(layerRuntime)
```

Comment:
I'm not a huge fan of this init, it feels an awful lot like the check() function that we tried to remove. Maybe it's required in this case? I did mention a possible alternative in another comment where the set variable function calls setLength, but I guess the issue here is this layer don't have the a set variable function to call setLength since this layer has no variables?

A thought, does it work to have an empty set variable functions? For example, maybe this checksum layer implementation could do:
```scala
def setLayerVariableParameters(): Unit = {
  setLength(20)
}
```

So the documentation for ChecksumLayer api says all implementations must implement a setLayerVariableParameters function and must call setLength(). If the layer has no variables, the function can have an empty parameter list.

So setLayerVariablParameters kindof becomes a layers `init()` function, and it can be parameterized by variables or not?